x/staking/client/cli/tx.go

```diff
@@ -1,26 +1,406 @@
+	cmd.Flags().String(FlagAmount, "0", "--amount=<amount>")
+	cmd.Flags().Uint64(FlagActivationEpoch, 0, "--activation-epoch=<activation-epoch>")
+
+	if err := cmd.MarkFlagRequired(FlagBlockNumber); err != nil {
+		logger.Error("SendValidatorJoinTx | MarkFlagRequired | FlagBlockNumber", "Error", err)
+	}
+	if err := cmd.MarkFlagRequired(FlagActivationEpoch); err != nil {
+		logger.Error("SendValidatorJoinTx | MarkFlagRequired | FlagActivationEpoch", "Error", err)
+	}
+	if err := cmd.MarkFlagRequired(FlagAmount); err != nil {
+		logger.Error("SendValidatorJoinTx | MarkFlagRequired | FlagAmount", "Error", err)
+	}
+	if err := cmd.MarkFlagRequired(FlagSignerPubkey); err != nil {
+		logger.Error("SendValidatorJoinTx | MarkFlagRequired | FlagSignerPubkey", "Error", err)
+	}
+	if err := cmd.MarkFlagRequired(FlagTxHash); err != nil {
+		logger.Error("SendValidatorJoinTx | MarkFlagRequired | FlagTxHash", "Error", err)
+	}
+	return cmd
+}
+
+// SendValidatorUpdateTx send validator update transaction
+func SendSignerUpdateTxCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "signer-update",
+		Short: "Update signer for a validator",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			// get proposer
+			proposer := hmTypes.HexToAccAddress(viper.GetString(FlagProposerAddress))
+			if proposer.Empty() {
+				proposer = helper.GetFromAddress(clientCtx)
+			}
+
+			validator := viper.GetUint64(FlagValidatorID)
+			if validator == 0 {
+				return fmt.Errorf("validator ID cannot be 0")
+			}
+
+			pubkeyStr := viper.GetString(FlagNewSignerPubkey)
+			if pubkeyStr == "" {
+				return fmt.Errorf("Pubkey has to be supplied")
+			}
+
+			pubkeyBytes, err := hex.DecodeString(pubkeyStr)
+			if err != nil {
+				return err
+			}
+			pubkey := hmTypes.NewPubKey(pubkeyBytes)
+
+			txhash := viper.GetString(FlagTxHash)
+			if txhash == "" {
+				return fmt.Errorf("transaction hash has to be supplied")
+			}
+
+			msg := types.NewMsgSignerUpdate(
+				proposer,
+				validator,
+				pubkey,
+				hmTypes.HexToHeimdallHash(txhash),
+				viper.GetUint64(FlagLogIndex),
+				viper.GetUint64(FlagBlockNumber),
+				viper.GetUint64(FlagNonce),
+			)
+
+			// broadcast messages
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+		},
+	}
+
+	cmd.Flags().StringP(FlagProposerAddress, "p", "", "--proposer=<proposer-address>")
+	cmd.Flags().Uint64(FlagValidatorID, 0, "--id=<validator-id>")
+	cmd.Flags().String(FlagNewSignerPubkey, "", "--new-pubkey=<new-signer-pubkey>")
+	cmd.Flags().String(FlagTxHash, "", "--tx-hash=<transaction-hash>")
+	cmd.Flags().Uint64(FlagLogIndex, 0, "--log-index=<log-index>")
+	cmd.Flags().Uint64(FlagBlockNumber, 0, "--block-number=<block-number>")
+	cmd.Flags().Int(FlagNonce, 0, "--nonce=<nonce>")
+
+	if err := cmd.MarkFlagRequired(FlagValidatorID); err != nil {
```

Comment:
Do we really need to handle error here? Personally I don't like having these error handlers for `cmd` methods.

Reply:
wdyt @jdkanani

Reply:
Yeah, we don't have to handle errors here.